### PR TITLE
python3 compatibility

### PR DESCRIPTION
**Pull request by petschki**
there are 2 tests failing but i'll tackle them down today ...

### CODE REVIEW

**Comment by petschki** [comment]
@pbauer @MrTango this fixes #27

File: plone/outputfilters/filters/example.py [hunk 1]
```diff
@@ -18,4 +18,4 @@ def is_enabled(self):
     pattern = re.compile(r'--')
 
     def __call__(self, data):
-        return self.pattern.sub(u'—', data)
+        return self.pattern.sub(u'\xe2\x80\x94', data)
```
Comment (petschki) [inline plone/outputfilters/filters/example.py]: @pbauer maybe you have a quicktip how to solve this doctest? py3 passes while py2 gives the following
bildschirmfoto 2018-07-16 um 14 07 47

Comment (pbauer) [reply plone/outputfilters/filters/example.py]: This is maybe due to remaining issues in PortalTransforms (https://github.com/plone/Products.CMFPlone/projects/4#card-11315725). If not you can use the Py23DocChecker-pattern to work around it.

**Comment by petschki** [comment]
@pbauer test are passing in py2 and py3 now ... hopefully I gain more experience to py2/py3 non ascii string fighting at the Halle Sprint, but for now I've ended up splitting the README doctest files in py3 (https://github.com/plone/plone.outputfilters/blob/python3-therp/plone/outputfilters/README.rst) and py2 (https://github.com/plone/plone.outputfilters/blob/python3-therp/plone/outputfilters/tests/README_py2.rst)

**Comment by pbauer** [comment]
I merged this since I wanted to be able to review the whole pr for py3. But I just realized that there are merge-conflicts since https://github.com/plone/plone.outputfilters/pull/28 and https://github.com/plone/plone.outputfilters/pull/29 by @tomgross were merged. 

@petschki can you please try to rebase the python3 branch on master and resolve these conflicts? You'd be much faster than I since you already know what the code is doing...

**Comment by petschki** [comment]
Yep. Will do tomorrow.

**Comment by petschki** [comment]
@tomgross @pbauer I've rebased python3 branch on master and resolved the conflicts. But now there is one test failing regarding resolving absolute `/resolveuid/xxx` links ... I'll fix them within the next minutes and then we're ready... hold on

**Pull request merged**